tests/serve_request.rs

```diff
@@ -152,6 +152,33 @@ fn serves_requests_no_hidden_files_without_flag(server: TestServer) -> Result<()
     Ok(())
 }
 
+#[rstest]
+#[case(false, server(None::<&str>))]
+#[case(true, server(&["--no-symlinks"]))]
+fn serves_requests_nested_in_symlinks(
+    #[case] no_symlinks: bool,
+    #[case] server: TestServer,
+) -> Result<(), Error> {
+    let expected_status = if no_symlinks {
+        StatusCode::NOT_FOUND
+    } else {
+        StatusCode::OK
+    };
```

Comment:
Put these into the `cases` so you don't have to have this conditional here. It also makes it more obvious as to what we expect in each case.